packages/vkui/src/components/DateInput/DateInput.tsx

```diff
@@ -175,7 +175,14 @@ export const DateInput = ({
       }
 
       if (isMatch(formattedValue, mask)) {
-        onChange?.(parse(formattedValue, mask, value ?? new Date()));
+        onChange?.(
+          parse(
+            formattedValue,
+            mask,
+            value ??
+              new Date(enableTime ? new Date().setSeconds(0, 0) : new Date().setHours(0, 0, 0, 0)),
+          ),
+        );
```

Comment:
```suggestion
        const now = new Date();
        onChange?.(
          parse(formattedValue, mask, value ?? (enableTime ? startOfMinute(now) : startOfDay(now))),
        );
```

Только надо не забыть импортнуть функции